### PR TITLE
Kth smallest element in an array 

### PR DESCRIPTION
**Pull request by hw0105**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
Added the code for finding the kth smallest element in an array in cpp. #4413

### Checklist

- [ yes] I've read the contribution guidelines.
- [yes ] I've checked the issue list before deciding what to submit.
- [yes ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#4413

### CODE REVIEW

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces

**Review by iamrajiv — APPROVED** [review]

File: C-Plus-Plus/heap/KthSmallestElement.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+using namespace std;
+/*
+  Following is the code to find the Kth smallest element in an array
+  Approach:
+      1.  Maintain a max heap of size k.
+      2.  Traverse the array and remove the top element if it becomes
+          greater than the current element of the array.
+      3.  The top element of the heap after the traversal of the array
+          is the Kth smallest element of the array.
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/heap/KthSmallestElement.cpp:9]: Shift your approach above the header

File: C-Plus-Plus/heap/KthSmallestElement.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                k=4
+                Kth Smallest Element: 7
+    Test Case 2:
+                n=10
+                arr[]=120,110,115,130,46,15,23,48,69,75
+                k=10
+                Kth Smallest Element: 130
+      Time Complexity: O(nlogk)
+      Space Complexity: O(k)
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/heap/KthSmallestElement.cpp]: Code produces output like that rewrite it.
Capture

Remove the extra test case one is enough

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/heap/KthSmallestElement.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+}
+/*
+    Test Case 1:
+                Enter the size of the array: 8
+                Enter k: 4
+                Enter array elements: 7 8 9 5 7 2 4 10
+                The Kth Smallest Element: 7
+      Time Complexity: O(nlogk)
+      Space Complexity: O(k)
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/heap/KthSmallestElement.cpp]: Add a new line at the end 
Remove the space before the testcase, input and output Rewrite it like that,
```
/*
Sample Input:

Enter the size of the array: 8
Enter k: 4
Enter array elements: 7 8 9 5 7 2 4 10

Sample Output:
The Kth Smallest Element: 7

Time Complexity: O(nlogk)
Space Complexity: O(k)
*/
```

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rest looks good to merge

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**